### ISSUE TITLE
sshfp.json registration

### ISSUE THREAD

**Issue by bitcynth** [protocol-registries/well-known-uris#2]
Please confirm that:

* [x] You have read and understood the [requirements for registration](https://tools.ietf.org/html/rfc8615#section-3.1).
* [x] You have checked [the registry](https://www.iana.org/assignments/well-known-uris/) and found no current value that meets your needs.
* [x] Your specification reference URL is stable; ideally, managed by a widely-recognised standards development organisation (e.g., published as an RFC). Otherwise, please give additional information below.

If so, please provide the details of your request below.

Enter the details of the link relation type below:

* URI suffix: sshfp.json
* Change controller: Cynthia Revström, [https://cynthia.re](https://cynthia.re), [d6035904-6d06-4ef2-861b-81090738b440@email.cynthia.re](mailto:d6035904-6d06-4ef2-861b-81090738b440@email.cynthia.re)
* Specification document(s): https://cynthia.re/docs/sshfpjson

Any additional information (this will not be included in the registry)?
N/A

**Comment by linuxgemini** [comment]
I am asking this because Section 5.2 of RFC8615 defines Well-Known URIs on multiple protocols (And the RFC itself defines for `http` _and_ `https`).

How is trust achieved on the usage of this file on any protocols outside `https`? The SSHFP DNS record definition on RFC4255 specifies DNSSEC for verification.

Although DNSSEC on its own has its own questions for trust, I do think that a trust method should be documented as well.

**Comment by mnot** [comment]
Hello,

There are a few issues with this request -

1. The URL for the specification document appears to be a personal Web site. Experience has shown that these URLs aren't stable in the long run.
2. The specification is very brief, and doesn't address how to infer the relationship between a given port on a given host serving HTTP(S) and one serving SSH. E.g., should one only infer a relationship when the hosts match, or is it acceptable to get a key for foo.example.org from www.example.org? What if the ports are not default ports? Etc. 
3. As @linuxgemini points out, there are issues involved with making this available over HTTP. Have you considered constraining it to just HTTPS?
4. Experience shows that specifications have a much better chance of succeeding when they are community efforts; merely registering a value doesn't make it more likely. Have you talked to people about this (e.g., on the [secdispatch list](https://www.ietf.org/mailman/listinfo/secdispatch))?
5. The RFC discourages well-known URIs from constraining the response content type, which your spec does (and this is implied by the `.json` suffix as well). This disallows content negotiation and the possibility of alternative formats in the future.

I'd encourage you to consider the above and adjust the proposal as necessary. That said, the only registration-related concern is with (1) above. So, I'm inclined to make this a *provisional* registration and review it in six months to see if it should be kept.

Does that work for you?

**Comment by bitcynth** [comment]
TL;DR I will get back to you within a few days with an update.

1. I will look at potential other ways to publish this specification.

2. This is addressed, however I acknowledge that it is not very well documented, I will improve the documentation.

3. Good point, this is was the plan but for whatever reason didn't get written down.

4. This has been discussed with other people looking for something like this.

5. I will look into this and get back to you within a few days.

**Comment by bitcynth** [comment]
Hello,
I have now updated the specs slightly and they are now hosted on GitHub at the following url: https://sshfp.github.io/sshfp.html

**Comment by mnot** [comment]
Registration approved as provisional, with the following details:

* URI suffix: “sshfp”
* Change controller: Cynthia Revström, https://cynthia.re, d6035904-6d06-4ef2-861b-81090738b440@email.cynthia.re
* Specification: this document

Please be aware that if it's judged to be in misuse in the future, it can be removed from the registry.
